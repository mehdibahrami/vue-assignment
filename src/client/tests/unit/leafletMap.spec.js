import { shallowMount } from "@vue/test-utils";
import Map from "@/components/maps/LeafletMap.vue";

describe("LeafletMap", () => {
  let wrapper;

  it("renders correctly", () => {
    // shallow mounting the main component
    wrapper = shallowMount(Map, {
      global: {
        renderStubDefaultSlot: true,
      },
      props: {
        lat: 52.093448638916016,
        lng: 5.117378234863281,
        soc: 72.8,
      },
    });

    expect(wrapper.findComponent({ name: "LMap" }).exists()).toBeTruthy();
    expect(wrapper.findComponent({ name: "LTileLayer" }).exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });
});
