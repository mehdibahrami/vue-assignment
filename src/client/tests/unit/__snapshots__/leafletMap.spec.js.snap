// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeafletMap renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <!-- Leaflet map -->
  <l-map-stub
    center="LatLng(52.093449, 5.117378)"
    class="map"
    crs="EPSG3857"
    noblockinganimations="false"
    options="[object Object]"
    usegloballeaflet="false"
    worldcopyjump="false"
    zoom="15"
  >
    <!-- Tile layer -->
    <l-tile-layer-stub
      detectretina="false"
      minzoom="0"
      nowrap="false"
      opacity="1"
      options="[object Object]"
      pane="tilePane"
      subdomains="abc"
      tilesize="256"
      tms="false"
      url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
      visible="true"
      zindex="1"
    />
    <!-- Marker -->
    <l-marker-stub
      draggable="false"
      icon="[object Object]"
      latlng="LatLng(52.093449, 5.117378)"
      options="[object Object]"
      pane="markerPane"
      visible="true"
    />
  </l-map-stub>
  
</div>
`;
